### ISSUE TITLE
ugrep returns duplicate results due to symlinks

### ISSUE THREAD

**Issue by carlwgeorge** [Genivia/ugrep#217]
I noticed today that ugrep defaults to `-R` (de-referenced recursive) behavior if no file arguments or standard input are given.  This differs from other modern grep tools whose default behavior matches grep's `-r` (recursive) option.

```
$ mkdir foo bar baz
$ echo 'Hello world!' > foo/hello.txt
$ ln -s ../foo bar/
$ ln -s ../foo baz
$ ack world
foo/hello.txt
1:Hello world!
$ ag world
foo/hello.txt
1:Hello world!
$ rg world
foo/hello.txt
1:Hello world!
$ ug world
bar/foo/hello.txt:Hello world!
baz/foo/hello.txt:Hello world!
foo/hello.txt:Hello world!
```

The man page seems to confirm that this behavior is intentional.

```
If  no  FILE  arguments  are specified and standard input is read from a terminal,
recursive searches are performed as if -R is specified.
```

It is nice to see that ugrep supports both `-R` and `-r` flags (yay for grep compatiblity), but would you consider making the `-r` behavior the default?

**Comment by genivia-inc** [comment]
Thanks for the feedback!

Some time ago I got frustrated that files would not show up with grep tools, when I realized these were symlinked so `-R` would of course work. But it may not be intuitive to some that the default ignores symlinks. I think is worse to skip files rather then include them in the search. It will make the search a bit slower, but ugrep is fast enough to make the difference negligible in most cases. But, that's just my opinion.

I think this question is up to a vote by users, rather than authoritatively asserting what is best.
